src/prepare/runtime.ts

```diff
@@ -0,0 +1,71 @@
+import type { Nuxt } from '@nuxt/schema'
+import type { I18nNuxtContext } from '../context'
+import { i18nVirtualLoggerPlugin, RESOLVED_VIRTUAL_NUXT_I18N_LOGGER, VIRTUAL_NUXT_I18N_LOGGER } from '../virtual-logger'
+import { defu } from 'defu'
+import { addPlugin, addTemplate, addTypeTemplate } from '@nuxt/kit'
+import { generateTemplateNuxtI18nOptions } from '../template'
+import { generateI18nTypes, generateLoaderOptions, simplifyLocaleOptions } from '../gen'
+import { NUXT_I18N_TEMPLATE_OPTIONS_KEY } from '../constants'
+
+export function prepareRuntime(ctx: I18nNuxtContext, nuxt: Nuxt) {
+  const { isDev: dev, isSSG, localeCodes, localeInfo, normalizedLocales, options, resolver, vueI18nConfigPaths } = ctx
+  // for core plugin
+  addPlugin(resolver.resolve('./runtime/plugins/i18n'))
+  addPlugin(resolver.resolve('./runtime/plugins/switch-locale-path-ssr'))
+
+  // for composables
+  nuxt.options.alias['#i18n'] = resolver.resolve('./runtime/composables/index')
+  nuxt.options.build.transpile.push('#i18n')
+  nuxt.options.build.transpile.push(VIRTUAL_NUXT_I18N_LOGGER)
+
+  const genTemplate = (isServer: boolean, lazy?: boolean) => {
+    const nuxtI18nOptions = defu({}, options)
+    // override `lazy` options
+    if (lazy != null) {
+      nuxtI18nOptions.lazy = lazy
+    }
+    return generateTemplateNuxtI18nOptions({
+      ...generateLoaderOptions(nuxt, {
+        vueI18nConfigPaths,
+        localeInfo,
+        nuxtI18nOptions,
+        isServer
+      }),
+      localeCodes,
+      normalizedLocales,
+      dev,
+      isSSG,
+      parallelPlugin: options.parallelPlugin
+    })
+  }
+
+  ctx.genTemplate = genTemplate
+
+  nuxt.options.runtimeConfig.public.i18n.locales = simplifyLocaleOptions(nuxt, defu({}, options))
```

Comment:
I'm guessing we can't move this to `prepare/runtime-config.ts` right? 🤔

Reply:
oh, we can do what we want here

Reply:
we can move runtime config after runtime and then we can move it: do you want me to move it?